SB-235 (refactor) : 출고 요청서 목록 조회 응답 형식 변경

seob7 opened the pull request:
# SB-235 (refactor) : 출고 요청서 목록 조회 응답 형식 변경 #43

기존에 사용하던 응답 형식은 shipping 정보와 goods 정보를 다른 레벨로 분리하여 사용하였습니다. 
변경된 형식은 shipping & goods 를 하나의 레벨로 묶어 사용하도록 변경하였습니다.

## 목록 조회 변경 후 
```
{
    "result": {
        "resultCode": 200,
        "resultMessage": "성공",
        "resultDescription": "성공"
    },
    "body": {
        "shippingList": [
            {
                "shippingId": 0,
                "deliveryDate": "string",
                "deliveryAddress": "string",
                "status": "REGISTERED",
                "deliveryMan": null,
                "goods": [
                    {
                        "id": 0,
                        "name": "string",
                        "category": "SMALL_FURNITURE",
                        "quantity": 0,
                        "basicImages": [
                            {
                                "id": 0,
                                "serverName": "string",
                                "originalName": "string",
                                "imageUrl": "string",
                                "caption": "string",
                                "kind": "BASIC",
                                "goodsId": 0
                            }
                        ],
                        "faultImages": [
                            {
                                "id": 0,
                                "serverName": "string",
                                "originalName": "string",
                                "imageUrl": "string",
                                "caption": "string",
                                "kind": "BASIC",
                                "goodsId": 0
                            }
                        ]
                    },
                    {
                        "id": 15,
                        "name": "이름",
                        "category": "SMALL_FURNITURE",
                        "quantity": 2,
                        "basicImages": [
                            {
                                "id": 0,
                                "serverName": "string",
                                "originalName": "string",
                                "imageUrl": "string",
                                "caption": "string",
                                "kind": "BASIC",
                                "goodsId": 0
                            }
                        ],
                        "faultImages": [
                            {
                                "id": 0,
                                "serverName": "string",
                                "originalName": "string",
                                "imageUrl": "string",
                                "caption": "string",
                                "kind": "BASIC",
                                "goodsId": 0
                            }
                        ]
                    }
                ]
            },
            {
                "shippingId": 0,
                "deliveryDate": "string",
                "deliveryAddress": "string",
                "status": "PENDING",
                "deliveryMan": null,
                "goods": [
                    {
                        "id": 0,
                        "name": "이름",
                        "category": "SMALL_FURNITURE",
                        "quantity": 0,
                        "basicImages": [],
                        "faultImages": []
                    },
                    {
                        "id": 3,
                        "name": "string",
                        "category": "SMALL_FURNITURE",
                        "quantity": 0,
                        "basicImages": [],
                        "faultImages": []
                    }
                ]
            }
        ]
    }
}
```


## 상세 보기 변경 후
```
{
    "result": {
        "resultCode": 200,
        "resultMessage": "성공",
        "resultDescription": "성공"
    },
    "body": {
        "shippingId": 5,
        "deliveryDate": "2024-07-23T12:12:24.547",
        "deliveryAddress": "string",
        "status": "REGISTERED",
        "deliveryMan": null,
        "goods": [
            {
                "id": 0,
                "name": "string",
                "category": "SMALL_FURNITURE",
                "quantity": 0,
                "basicImages": [],
                "faultImages": []
            },
            {
                "id": 0,
                "name": "string",
                "category": "SMALL_FURNITURE",
                "quantity": 0,
                "basicImages": [],
                "faultImages": []
            }
        ]
    }
```




---

## ❌변경 전
```
{
  "result": {
    "resultCode": 0,
    "resultMessage": "string",
    "resultDescription": "string"
  },
  "body": {
    "shipping": [
      {
        "shipping": {
          "shippingId": 0,
          "deliveryDate": "2024-08-01T03:56:50.310Z",
          "deliveryAddress": "string",
          "status": "PENDING",
          "deliveryMan": 0
        },
        "goods": [
          {
            "id": 0,
            "name": "string",
            "category": "SMALL_FURNITURE",
            "quantity": 0,
            "basicImages": [
              {
                "id": 0,
                "serverName": "string",
                "originalName": "string",
                "imageUrl": "string",
                "caption": "string",
                "kind": "BASIC",
                "goodsId": 0
              }
            ],
            "faultImages": [
              {
                "id": 0,
                "serverName": "string",
                "originalName": "string",
                "imageUrl": "string",
                "caption": "string",
                "kind": "BASIC",
                "goodsId": 0
              }
            ]
          }
        ]
      }
    ]
  }
}
```

## ❌변경 후
```
{
    "result": {
        "resultCode": 200,
        "resultMessage": "성공",
        "resultDescription": "성공"
    },
    "body": [
        {
            "shippingId": 0,
            "deliveryDate": "2024-07-23T12:12:24.547",
            "deliveryAddress": "string",
            "status": "REGISTERED",
            "deliveryMan": null,
            "goods": [
                {
                    "id": 0,
                    "name": "string",
                    "category": "SMALL_FURNITURE",
                    "quantity": 0,
                    "basicImages": [
                        {
                            "id": 13,
                            "serverName": "string",
                            "originalName": "string",
                            "imageUrl": "string",
                            "caption": "string",
                            "kind": "BASIC",
                            "goodsId": 0
                        }
                    ],
                    "faultImages": [
		                    {
				                    "id": 13,
                            "serverName": "string",
                            "originalName": "string",
                            "imageUrl": "string",
                            "caption": "string",
                            "kind": "BASIC",
                            "goodsId": 0
	
		                    }
                    ]
                }
            ]
        },
         {
            "shippingId": 0,
            "deliveryDate": "2024-07-23T12:12:24.547",
            "deliveryAddress": "string",
            "status": "REGISTERED",
            "deliveryMan": null,
            "goods": [
                {
                    "id": 0,
                    "name": "string",
                    "category": "SMALL_FURNITURE",
                    "quantity": 0,
                    "basicImages": [
                        {
                            "id": 13,
                            "serverName": "string",
                            "originalName": "string",
                            "imageUrl": "string",
                            "caption": "string",
                            "kind": "BASIC",
                            "goodsId": 0
                        }
                    ],
                    "faultImages": [
		                    {
				                    "id": 13,
                            "serverName": "string",
                            "originalName": "string",
                            "imageUrl": "string",
                            "caption": "string",
                            "kind": "BASIC",
                            "goodsId": 0
	
		                    }
                    ]
                }
            ]
        }
    ]
}
```


1. ShippingBusiness
- List 를 반환하도록 변경
2. ShippingListResponse
3. ShippingApiController
4. ShippingConverter

seob7 committed:
SB-235 (refactor) : 출고 요청서 목록 조회 응답 형식 변경

SB-235 (refactor) : 출고 요청서 목록 조회 응답 형식 변경

1. ShippingBusiness
- List 를 반환하도록 변경
2. ShippingListResponse
3. ShippingApiController
4. ShippingConverter

jeongHunE commented:
제가 잘못 확인한 부분이 있어서 정정하고자 comment 남깁니다.

```
{
  "result": {
    "resultCode": 200,
    "resultMessage": "성공",
    "resultDescription": "성공"
  },
  "body": {
    "shippingList": [
      {
        "shippingId": 5,
        "deliveryDate": "2024-07-23T12:12:24.547",
        "deliveryAddress": "string",
        "status": "REGISTERED",
        "deliveryMan": null,
        "goods": [
          {
            "id": 0,
            "name": "string",
            "category": "SMALL_FURNITURE",
            "quantity": 0,
            "basicImages": [
              {
                "id": 13,
                "serverName": "string",
                "originalName": "string",
                "imageUrl": "string",
                "caption": "string",
                "kind": "BASIC",
                "goodsId": 0
              }
            ],
            "faultImages": [
              {
                "id": 13,
                "serverName": "string",
                "originalName": "string",
                "imageUrl": "string",
                "caption": "string",
                "kind": "BASIC",
                "goodsId": 0
              }
            ]
          }
        ]
      }
    ]
  }
}
```

`body`에 바로 배열을 넣는 것 대신 배열을 object로 한번 감싸 주시면 좋을 것 같습니다.

seob7 committed:
SB-235 (refactor) : 출고 요청서 목록 & 상세 보기 응답 형식 변경

SB-235 (refactor) : 출고 요청서 목록 & 상세 보기 응답 형식 변경

1. ShippinResponse 제거 -> ShippingDetailResponse 로 대체
2. ShippingListResponse
- List<ShippingDetailResponse> 로 변경
3. ShippingApiController
- 요청서 목록 조회 반환 형식 변경
4. ShippingConverter
- 사용하지 않는 `toResponse` 메서드 제거
5. ShippingBusiness
- 목록 보기 & 상세 보기 로직 및 반환값 변경
